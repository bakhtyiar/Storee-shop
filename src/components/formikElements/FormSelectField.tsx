import React from "react";
import {Form, InputGroup} from "react-bootstrap";
import {Field} from "formik";

const FormSelectField = ({
    as,
    md,
    controlId,
    label,
    name,
    type,
    inputGroupPrepend,
    children,
    defaultValue
}: any) => {
    return (
        <Field name={name}>
            {({
                field,
                form
            }: any) => {
                const isValid = !form.errors[field.name];
                const isInvalid = form.touched[field.name] && !isValid;
                return (
                    <Form.Group as={as} md={md} controlId={controlId} data-testid="form-select-field">
                        
                        <Form.Label>{label}</Form.Label>
                        
                        <InputGroup>
                            {inputGroupPrepend}
                            
                            <Form.Select
                                defaultValue={defaultValue}
                                className={'rounded'}
                                {...field}
                                type={type}
                                isValid={form.touched[field.name] && isValid}
                                isInvalid={isInvalid}
                                feedback={form.errors[field.name]}
                                as="select"
                            >
                                {children}
                            </Form.Select>
                            
                            <Form.Control.Feedback type="invalid">
                                {form.errors[field.name]}
                            </Form.Control.Feedback>
                        </InputGroup>
                    </Form.Group>
                );
            }}
        </Field>
    );
};

FormSelectField.defaultProps = {
    type: "select",
    inputGroupPrepend: null
};

export default FormSelectField;
