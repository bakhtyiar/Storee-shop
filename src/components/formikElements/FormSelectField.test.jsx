import {screen} from "@testing-library/react";
import {renderWithFormikContext} from "../../tests/helpers/renderWithFormikContext";
import FormSelectField from "./FormSelectField";

describe("Basic tests", () => {
    test("Match snapshot", () => {
        renderWithFormikContext(<FormSelectField/>);
        expect(screen.getByTestId("form-select-field")).toMatchSnapshot();
    })
})