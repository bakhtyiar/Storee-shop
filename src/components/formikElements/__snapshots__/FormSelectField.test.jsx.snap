// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests Match snapshot 1`] = `
<div
  data-testid="form-select-field"
>
  <label
    class="form-label"
  />
  <div
    class="input-group"
  >
    <select
      as="select"
      class="rounded form-select"
      type="select"
    />
    <div
      class="invalid-feedback"
    />
  </div>
</div>
`;
